### PR TITLE
fix maximum amt

### PR DESCRIPTION
**Pull request by shawnxie999**
### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Performance (increase or change in throughput and/or latency)
- [ ] Tests (you added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation update
- [ ] Chore (no impact to binary, e.g. `.gitignore`, formatting, dropping support for older tooling)
- [ ] Release

### API Impact

- [ ] Public API: New feature (new methods and/or new fields)
- [ ] Public API: Breaking change (in general, breaking changes should only impact the next api_version)
- [ ] `libxrpl` change (any change that may affect `libxrpl` or dependents of `libxrpl`)
- [ ] Peer protocol change (must be backward compatible or bump the peer protocol version)

### CODE REVIEW

File: src/xrpld/ledger/detail/View.cpp [hunk 1]
```diff
@@ -1353,6 +1353,18 @@ rippleSend(
...
+        // if sender is issuer, check that the new OutstandingAmount will not
+        // exceed MaximumAmount
+        if (uSenderID == issuer)
+        {
+            auto const mptID = keylet::mptIssuance(saAmount.issue().getMptID());
+            auto const sle = view.peek(mptID);
+
+            if (sle->getFieldU64(sfOutstandingAmount) + saAmount.value() >
+                (*sle)[~sfMaximumAmount].value_or(maxMPTokenAmount))
+                return tecMPT_MAX_AMOUNT_EXCEEDED;
```
Comment (gregtatcam) [inline src/xrpld/ledger/detail/View.cpp:1367]: Should add a unit-test for this.

Comment (shawnxie999) [reply src/xrpld/ledger/detail/View.cpp:1367]: we already have existing unit tests that tests if the issuer issues out more than the max. the unit test i added covered the scenario reproduced in QA that currently exhibits the wrong behavior

Comment (gregtatcam) [reply src/xrpld/ledger/detail/View.cpp:1367]: Isn't this a different execution path?

Comment (shawnxie999) [reply src/xrpld/ledger/detail/View.cpp:1367]: All the existing test cases hit this new if-check in `rippleSend`. The problem found is that during holder to holder payment, `rippleCredit` is called twice from **1) holder to issuer** and then **2)issuer to holder**.  The current problem is that in the second `rippleCredit` call from issuer to holder, the code think the issuer is trying to issue more than the MaximumAmount. This is why I moved the check in the `rippleCredit` to its caller `rippleSend`.

**Pull request merged**